### PR TITLE
release-notes.txt: add 2023.01 release notes

### PR DESCRIPTION
**Pull request by kaspar030**
### Contribution description

Here are the auto-generated release-notes. Please take a look!

### CODE REVIEW

**Comment by kaspar030** [comment]
@RIOT-OS/maintainers this is closing in fast, any speak out now or be silent forever!

File: release-notes.txt [hunk 1]
```diff
@@ -1,3 +1,548 @@
...
+Things (IoT) devices. This release builds on top of the hard work and
+dedication of our developers, users, and contributors, and we would like to
+extend a heartfelt thank you to everyone who has been a part of our journey.
+Here's to another decade of innovation and growth!
+
+Noteworthy changes the 2023.01 release includes, among many fixes and little
+things:
+
+- ztimer_ondemand: the next step towards seamless and automatic power management
+- the Rust examples now default to stable Rust
```
Comment (bergzand) [inline release-notes.txt:33]: Nitpicks!

```suggestion
- ztimer_ondemand: The next step towards seamless and automatic power management.
- The Rust examples now default to stable Rust.
```

File: release-notes.txt [hunk 2]
```diff
@@ -1,3 +1,544 @@
...
+---
+
++ cpu/saml21: add interaction with pm_layered for peripheral drivers (#18821)
++ cpu/stm32: add unused backup RAM as extra heap (#18921)
++ debug_irq_disable: add module to debug time spent in irq_disable (#18795)
+* boards/native: Remove macOS as native target (#18735)
+* cpu/efm32: preparing introduction of Gecko Series 2 (#18764)
+* cpu/saml21/pm: allow blocking IDLE mode (#18825)
+* cpu/stm32: improvement of USB device driver selection (#18787)
+* drivers/usbdev_synopsys_dwc2: small fixes (#18759)
```
Comment (gschorcht) [inline release-notes.txt:141]: Should be moved to section `Bug fixes`.

Comment (gschorcht) [reply release-notes.txt:141]: Sorry for jumping in so late.

Comment (kaspar030) [reply release-notes.txt]: done!

**Review by bergzand — APPROVED** [review]
Now with an ack :)

bors merge

**Comment by kaspar030** [comment]
bors cancel
bors merge

**Pull request merged**